### PR TITLE
Add filters in multiple template files

### LINKED ISSUE #3320: Additional Filters in Templates
Hey guys. Throughout the templates, there are quite a few headers, messages, etc. that are hard coded in without any filters to change the text. This means users have to override the template (overkill) or use jQuery (no) to change the text. I think adding filters for these things would be great. I'd be happy to submit them if you let me know the proper way you like to receive pull requests (I'm new at PR's so bear with me).

### PR DESCRIPTION
**Pull request by EngageWP**
This PR addresses and can close #3320. It contains 22 new filters across 4 templates: checkout_cart.php, history-downloads.php, history-purchases.php and shortcode-login.php. This allows EDD users to change things like table headers, labels, etc. on some of the more prominent templates (i.e. cart/checkout page, account page with purchase/download history shortcodes and login page) without having to completely override the templates from their theme, which is overkill in my opinion. I'm committing these additions now and I'll continue to add new ones in the near future. Here are example uses:

``` php
/*
Add filters in checkout_cart.php template
*************************************************************/

/**
 *  Filter the <th> text for the "Item Name"
 *  @return (string)
 */
add_filter( 'edd_cart_table_item_name_label', 'rv_filter_cart_item_name_label' );
function rv_filter_cart_item_name_label() {
    return __( 'Download Name', 'edd' );
}

/**
 *  Filter the <th> text for the "Item Price"
 *  @return (string)
 */
add_filter( 'edd_cart_table_item_price_label', 'rv_filter_cart_item_price_label' );
function rv_filter_cart_item_price_label() {
    return __( 'Download Price', 'edd' );
}

/**
 *  Filter the <th> text for the "Actions"
 *  @return (string)
 */
add_filter( 'edd_cart_table_actions_label', 'rv_filter_cart_item_actions_label' );
function rv_filter_cart_item_actions_label() {
    return __( 'Cart Actions', 'edd' );
}

/**
 *  Filter the "Remove" link to remove an item from cart
 *  @return (string)
 */
add_filter( 'edd_cart_table_remove_item_link', 'rv_filter_cart_remove_item_link' );
function rv_filter_cart_remove_item_link() {
    return sprintf( '<img src="%s" width="20" class="aligncenter">', trailingslashit( get_stylesheet_directory_uri() ) . 'images/error.jpg' );
}

/**
 *  Filter the "Remove" link to remove a fee from cart
 *  @return (string)
 */
add_filter( 'edd_cart_table_remove_fee_link', 'rv_filter_cart_remove_fee_link' );
function rv_filter_cart_remove_fee_link() {
    return __( 'Delete', 'edd' );
}

/**
 *  Filter the "Subtotal:" label when taxes are used
 *  @return (string)
 */
add_filter( 'edd_cart_table_subtotal_label', 'rv_filter_cart_subtotal_label' );
function rv_filter_cart_subtotal_label() {
    return __( 'Before Tax:', 'edd' );
}

/**
 *  Filter the "Tax:" label when taxes are used
 *  @return (string)
 */
add_filter( 'edd_cart_table_tax_amount_label', 'rv_filter_cart_tax_label' );
function rv_filter_cart_tax_label() {
    return __( 'Tax Due:', 'edd' );
}

/**
 *  Filter the "Total:" label
 *  @return (string)
 */
add_filter( 'edd_cart_table_total_label', 'rv_filter_cart_total_label' );
function rv_filter_cart_total_label() {
    return __( 'Total Due:', 'edd' );
}



/*
Add filters for [download_history] shortcode (history-downloads.php template)
******************************************************************************/

/**
 *  Filter the "Download Name" header
 *  @return (string)
 */
add_filter( 'edd_download_history_download_name_label', 'rv_filter_download_history_download_name_label' );
function rv_filter_download_history_download_name_label() {
    return __( 'Item Name', 'edd' );
}

/**
 *  Filter the "Files" header
 *  @return (string)
 */
add_filter( 'edd_download_history_files_label', 'rv_filter_download_history_files_label' );
function rv_filter_download_history_files_label() {
    return __( 'Download Files', 'edd' );
}

/**
 *  Filter the "No downloadable files found." notice
 *  @return (string)
 */
add_filter( 'edd_download_history_no_files_found_notice', 'rv_filter_download_history_no_files_found_notice' );
function rv_filter_download_history_no_files_found_notice() {
    return __( 'No files to download', 'edd' );
}

/**
 *  Filter the "You have not purchased any downloads" notice
 *  @return (string)
 */
add_filter( 'edd_download_history_no_purchases_notice', 'rv_filter_download_history_no_purchases_notice' );
function rv_filter_download_history_no_purchases_notice() {
    return __( 'You have not purchased anything yet.', 'edd' );
}


/*
Add filters for [purchase_history] shortcode (history-purchases.php template)
******************************************************************************/

/**
 *  Filter the "ID" table header
 *  @return (string)
 */
add_filter( 'edd_purchase_history_table_purchase_id_label', 'rv_filter_purchase_history_id_label' );
function rv_filter_purchase_history_id_label() {
    return __( 'Order No.', 'edd' );
}

/**
 *  Filter the "Date" table header
 *  @return (string)
 */
add_filter( 'edd_purchase_history_table_purchase_date_label', 'rv_filter_purchase_history_date_label' );
function rv_filter_purchase_history_date_label() {
    return __( 'Bought On', 'edd' );
}

/**
 *  Filter the "Amount" table header
 *  @return (string)
 */
add_filter( 'edd_purchase_history_table_purchase_amount_label', 'rv_filter_purchase_history_amount_label' );
function rv_filter_purchase_history_amount_label() {
    return __( 'Paid', 'edd' );
}

/**
 *  Filter the "Details" table header
 *  @return (string)
 */
add_filter( 'edd_purchase_history_table_purchase_details_label', 'rv_filter_purchase_history_details_label' );
function rv_filter_purchase_history_details_label() {
    return __( 'Order Details', 'edd' );
}

/*
Add filters for [edd_login] shortcode (shortcode-login.php template)
******************************************************************************/

/**
 *  Filter the "Log into Your Account" form header
 *  @return (string)
 */
add_filter( 'edd_login_form_header', 'rv_filter_login_form_header' );
function rv_filter_login_form_header() {
    return __( 'Customer Login', 'edd' );
}

/**
 *  Filter the "Username" input label
 *  @return (string)
 */
add_filter( 'edd_login_form_username_input_label', 'rv_filter_login_form_username_label' );
function rv_filter_login_form_username_label() {
    return __( 'Enter your username', 'edd' );
}

/**
 *  Filter the "Password" input label
 *  @return (string)
 */
add_filter( 'edd_login_form_password_input_label', 'rv_filter_login_form_password_label' );
function rv_filter_login_form_password_label() {
    return __( 'Enter your password', 'edd' );
}

/**
 *  Filter the "Log In" button text
 *  @return (string)
 */
add_filter( 'edd_login_form_login_button_text', 'rv_filter_login_form_log_in_button_text' );
function rv_filter_login_form_log_in_button_text() {
    return __( 'Log me in', 'edd' );
}

/**
 *  Filter the "Lost Password?" link text
 *  @return (string)
 */
add_filter( 'edd_login_form_lost_password_link_text', 'rv_filter_login_form_lost_password_text' );
function rv_filter_login_form_lost_password_text() {
    return __( 'Lose something?', 'edd' );
}

/**
 *  Filter the "You are already logged in" notice
 *  @return (string)
 */
add_filter( 'edd_login_form_already_logged_in_notice', 'rv_filter_login_form_already_logged_in_notice' );
function rv_filter_login_form_already_logged_in_notice() {
    return __( 'You already be logged in!', 'edd' );
}
```

### CODE REVIEW

File: templates/checkout_cart.php [hunk 1]
```diff
@@ -80,12 +80,12 @@
 			<tr class="edd_cart_footer_row edd_cart_subtotal_row"<?php if ( ! edd_is_cart_taxed() ) echo ' style="display:none;"'; ?>>
 				<?php do_action( 'edd_checkout_table_subtotal_first' ); ?>
 				<th colspan="<?php echo edd_checkout_cart_columns(); ?>" class="edd_cart_subtotal">
-					<?php _e( 'Subtotal', 'edd' ); ?>:&nbsp;<span class="edd_cart_subtotal_amount"><?php echo edd_cart_subtotal(); ?></span>
+					<?php echo apply_filters( 'edd_cart_table_subtotal_label', __( 'Subtotal:', 'edd' ) ); ?>&nbsp;<span class="edd_cart_subtotal_amount"><?php echo edd_cart_subtotal(); ?></span>
```
Comment (cklosowski) [inline templates/checkout_cart.php]: Looks like you've moved the `:` into the translated string, this needs to be outside of it.

File: templates/checkout_cart.php [hunk 2]
```diff
@@ -98,7 +98,7 @@
 			<tr class="edd_cart_footer_row edd_cart_tax_row"<?php if( ! edd_is_cart_taxed() ) echo ' style="display:none;"'; ?>>
 				<?php do_action( 'edd_checkout_table_tax_first' ); ?>
 				<th colspan="<?php echo edd_checkout_cart_columns(); ?>" class="edd_cart_tax">
-					<?php _e( 'Tax', 'edd' ); ?>:&nbsp;<span class="edd_cart_tax_amount" data-tax="<?php echo edd_get_cart_tax( false ); ?>"><?php echo esc_html( edd_cart_tax() ); ?></span>
+					<?php echo apply_filters( 'edd_cart_table_tax_amount_label', __( 'Tax:', 'edd' ) ); ?>&nbsp;<span class="edd_cart_tax_amount" data-tax="<?php echo edd_get_cart_tax( false ); ?>"><?php echo esc_html( edd_cart_tax() ); ?></span>
```
Comment (cklosowski) [inline templates/checkout_cart.php]: Same as above. `:` is now inside your translations string

**Comment by cklosowski** [comment]
There are multiple places where `:` punctuation marks have been moved within translation strings.

**Comment by EngageWP** [comment]
Good eye, Chris. My mistake.

**Comment by cklosowski** [comment]
@EngageWP Can you fix those two instances of `:` being in the translation strings, and I'll merge this.

**Comment by EngageWP** [comment]
Done

**Comment by cklosowski** [comment]
@pippinsplugins these look good to me. want to take a look?

**Comment by pippinsplugins** [comment]
@cklosowski @EngageWP Let's move discussion of this back to #3320. I'm not sure this should go in.

**Comment by pippinsplugins** [comment]
The related issue was closed as wontfix so closing this too.

**Pull request closed without merging**